src/python/pants/backend/visibility/rule_types.py

```diff
@@ -216,16 +270,16 @@ def get_action(
                         logger.debug(
                             f"{visibility_rule.action.name}: the `{target_type}` target "
                             f"{target_addr} dependency to a target at {path} by dependency rule "
-                            f"{visibility_rule} declared in {self.path}"
+                            f"{str(visibility_rule)!r} declared in {self.path}"
                         )
                     else:
                         logger.debug(
                             f"{visibility_rule.action.name}: dependency on the `{target_type}` "
                             f"target {target_addr} from a target at {path} by dependent rule "
-                            f"{visibility_rule} declared in {self.path}"
+                            f"{str(visibility_rule)!r} declared in {self.path}"
```

Comment:
not sure if we actually need these debug logs any more, given the richer context returned from `check_dependency_rules()`.